pkg/server/grpc.go

```diff
@@ -24,3 +29,40 @@ func (g *GRPC) Create(_ context.Context, req *customer.CreateRequest) (*customer
 
 	return &customer.Empty{}, nil
 }
+
+func NewGRPC(cfg *config.Cfg, store CustomerStore) (*GRPC, error) {
+	lis, err := net.Listen("tcp", cfg.Server.GRPC.Endpoint)
+	if err != nil {
+		return &GRPC{}, err
```

Comment:
O padrão Go neste caso é retornar nulo para o valor principal e o erro concreto no segundo parâmetro. O chamador verifica primeiro se não ocorreu um erro, para então continuar o processamento.

```suggestion
		return nil, err
```